src/celeritas/track/detail/TrackSortUtils.cu

```diff
@@ -96,13 +96,15 @@ void sort_impl(TrackSlots const& track_slots,
     CELER_DEVICE_CHECK_ERROR();
 }
 
-// PRE: get_action is sorted, i.e. i <= j ==> get_action(i) <=
-// get_action(j)
-template<class F>
-__device__ void
-tracks_per_action_impl(Span<ThreadId> offsets, size_type size, F&& get_action)
+// PRE: actions are sorted
```

Comment:
I guess I allowed the separators/comments to be skipped during the initial review, but 
 ```suggestion
//---------------------------------------------------------------------------//
/*!
 * Calculate thread boundaries based on action ID.
 * \pre actions are sorted
 */
```